server/services/approverCaseloadService.ts

```diff
@@ -0,0 +1,131 @@
+import CommunityService from './communityService'
+import PrisonerService from './prisonerService'
+import { DeliusRecord, ManagedCaseForApproval } from '../@types/managedCase'
+import LicenceStatus from '../enumeration/licenceStatus'
+import LicenceType from '../enumeration/licenceType'
+import { User } from '../@types/CvlUserDetails'
+import { LicenceSummaryApproverView } from '../@types/licenceApiClientTypes'
+import LicenceKind from '../enumeration/LicenceKind'
+import { filterCentralCaseload } from '../utils/utils'
+import { LicenceApiClient } from '../data'
+
+export default class ApproverCaseloadService {
+  constructor(
+    private readonly prisonerService: PrisonerService,
+    private readonly communityService: CommunityService,
+    private readonly licenceApiClient: LicenceApiClient
+  ) {}
+
+  async getApprovalNeeded(user: User, prisonCaseload: string[]): Promise<ManagedCaseForApproval[]> {
+    const licences: LicenceSummaryApproverView[] = await this.licenceApiClient.getLicencesForApproval(
+      filterCentralCaseload(prisonCaseload),
+      user
+    )
+    const caseLoad: ManagedCaseForApproval[] = await this.mapLicencesToOffenders(licences)
+    return this.mapResponsibleComsToCases(caseLoad)
+  }
+
+  async getRecentlyApproved(user: User, prisonCaseload: string[]): Promise<ManagedCaseForApproval[]> {
+    const licences: LicenceSummaryApproverView[] = await this.licenceApiClient.getLicencesRecentlyApproved(
+      filterCentralCaseload(prisonCaseload),
+      user
+    )
+    const caseLoad: ManagedCaseForApproval[] = await this.mapLicencesToOffenders(licences)
+    return this.mapResponsibleComsToCases(caseLoad)
+  }
+
+  private pairDeliusRecordsWithNomis = async (
```

Comment:
bit of copy paste from the caseload service. 

I think this is reasonable as a relatively short term fix - it feels like it's unlikely that there will be a change required that will affect all caseloads at the same time and its probably easier to reimplement this in the backend in isolation from the other caseloads

Reply:
I held off refactoring this further - there's some gnarly stuff in here